osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -259,6 +269,7 @@ protected sealed override void ApplyResultInternal(JudgementResult result)
 
                 updateScore();
             }
+            updateRank();
```

Comment:
At the point all of these extra `updateRank()` calls are being added, it sure *looks* like `updateRank()` and `updateScore()` should just be one method.

In fact if I squint at master code, I believe there may be at least one potential bug related to a missing `updateRank()` call:

https://github.com/ppy/osu/blob/63136311b585052304fd53e74c526efd9881a093/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L495

The theory goes that if this call happens to not change accuracy but *also* require a rank update, the second one will not happen.

Reply:
if it's possible to just move the `updateRank()` calls in to `updateScore()` and rename it to  `updateRankAndScore()`? so that the `updateRank` will be execute right after the score update.

Reply:
Yes that is pretty much what I was proposing. Call it `updateScore()`, which kinda already encompasses rank.

Reply:
Implement the change at the latest commit